connect/src/com/telenor/connect/ConnectSdk.java

```diff
@@ -487,4 +510,65 @@ public static synchronized void sdkReinitializeMobileConnect() {
         MobileConnectSdkProfile profile = (MobileConnectSdkProfile) sdkProfile;
         profile.deInitialize();
     }
+
+    public static ConnectivityManager getConnectivityManager() {
+        return connectivityManager;
+    }
+
+    public static Network getCellularNetwork() {
+        return cellularNetwork;
+    }
+
+    public static Network getWifiNetwork() {
+        return wifiNetwork;
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public static void initalizeCellularNetwork() {
+        NetworkRequest networkRequest = new NetworkRequest.Builder()
+                .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                .addTransportType(NetworkCapabilities.TRANSPORT_CELLULAR)
+                .build();
+        connectivityManager.requestNetwork(
+                networkRequest,
+                new ConnectivityManager.NetworkCallback() {
+                    @Override
+                    public void onAvailable(Network network) {
+                        cellularNetwork = network;
+                    }
+                }
+        );
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public static void initalizeWiFiNetwork() {
+        NetworkRequest networkRequest = new NetworkRequest.Builder()
+                .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
+                .build();
+        connectivityManager.requestNetwork(
+                networkRequest,
+                new ConnectivityManager.NetworkCallback() {
+                    @Override
+                    public void onAvailable(Network network) {
+                        wifiNetwork = network;
+                    }
+                }
+        );
+    }
+
+    /**
+     * Initialize components common to both Mobile Connect and ConnectID SDK profiles
+     */
+    public static synchronized void initializeCommonComponents() {
+        connectivityManager
+                = (ConnectivityManager) getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (null == connectivityManager) {
+            return;
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            initalizeCellularNetwork();
```

Comment:
Nit picking, but could you reverse this if, make it return from the method if it is true, and then put it at the top of the method (guard)? We don't need the `ConnectivityManager` if we're on an old version.